src/content/docs/en/guides/server-side-rendering.mdx

```diff
@@ -99,137 +139,64 @@ You can add any of the [official Astro adapter integrations](/en/guides/integrat
-
-# This is my static, pre-rendered page
-```
-
-And for an endpoint:
-
-```js title="src/pages/myendpoint.js" {1}
-export const prerender = true;
-
-export async function GET() {
-  return {
-    body: JSON.stringify({ message: `This is my static endpoint` }),
-  };
-}
-```
-
-### Opting out of pre-rendering
+    Note that different adapters may also have different configuration settings. Read each adapter's documentation, and apply any necessary config options to your chosen adapter in `astro.config.mjs`
 
-For a mostly static site configured as `output: hybrid`, add `export const prerender = false` to any files that should be server-rendered:
+## On-demand rendering features
 
-```js title="src/pages/randomnumber.js" {1}
-export const prerender = false;
+### HTML streaming
 
-export async function GET() {
-  let number = Math.random();
-  return {
-    body: JSON.stringify({ number, message: `Here's a random number: ${number}` }),
-  };
-}
-```
-
-### `Astro.request.headers`
-
-The headers for the request are available on `Astro.request.headers`. This works like the browser's [`Request.headers`](https://developer.mozilla.org/en-US/docs/Web/API/Request/headers). It is a [Headers](https://developer.mozilla.org/en-US/docs/Web/API/Headers) object, a Map-like object where you can retrieve headers such as the cookie.
-
-```astro title="src/pages/index.astro" {2}
----
-const cookie = Astro.request.headers.get('cookie');
-// ...
----
-<html>
-  <!-- Page here... -->
-</html>
-```
-
-### `Astro.request.method`
-
-The HTTP method used in the request is available as `Astro.request.method`. This works like the browser's [`Request.method`](https://developer.mozilla.org/en-US/docs/Web/API/Request/method). It returns the string representation of the HTTP method used in the request.
+With HTML streaming, a document is broken up into chunks, sent over the network in order, and rendered on the page in that order. In `server` or `hybrid` mode, Astro uses HTML streaming to send each component to the browser as it renders them. This makes sure the user sees your HTML as fast as possible, although network conditions can cause large documents to be downloaded slowly, and waiting for data fetches can block page rendering.
 
-```astro title="src/pages/index.astro"
----
-console.log(Astro.request.method) // GET (when navigated to in the browser)
----
-```
+<RecipeLinks slugs={["en/recipes/streaming-improve-page-performance"]}/>
 
-:::caution
-The features below are only available at the page level. (You can't use them inside of components, including layout components.)
+:::caution[HTML streaming]
+Features that modify the [Response headers](https://developer.mozilla.org/en-US/docs/Glossary/Response_header) are only available at the **page level** when HTML streaming is enabled. (You can't use them inside of components, including layout components.) By the time Astro runs your component code, it has already sent the Response headers and they cannot be modified.
```

Comment:
> when HTML streaming is enabled.

Is this the case. I don't know for sure, but I would be surprised if it is. Usually, the pattern has been that Astro maintains its streaming limitations even during static rendering.

Reply:
Ah, so my takeaway is it's *correct* but not limited to SSR.

So just confirming we still think that the SSR page is the right place for this. (Even if not limited to SSR, is this a place where people will more likely run into that limitation with what they're trying to do with on-demand rendering?)